.github/workflows/validate-pr-title.yaml

```diff
@@ -3,7 +3,7 @@ name: "Validate PR Title"
 # Ensure that the PR title conforms to the Conventional Commits and our choice of types and scopes, so that library version bumps can be detected automatically
 
 on:
-  pull_request_target:
+  pull_request:
```

Comment:
I'm afraid that the external PR titles cannot be validated with this change.

pr target: runs in the base branch context, has access to secrets, can therefore use the GH api to query the PR title.

pr: runs in the feature branch context, only has access to the github context.


It is possible that in our case, our specific config both would work, but please check carefully:
- pr title -- I'm not sure!
- pr constituent commit titles -- would not have access, but we don't care about those.
- pr labels -- we don't care about those.

If this change is actually OK, that is if the 3rd party action falls back to smth like `${{ github.event.pull_request.title }}`, then please change the permissions block 3 lines down.

Reply:
I picked Ben to review because I thought he was most likely to object and picked you to review because I thought you'd most likely find issues :)

This seemed to be working fine (I got an error when I had a bad title, and the error went away when I fixed it). But I do now weirdly get a lot of errors about this being cancelled. I assume that must be related, so I'll change this back and mark it as to be ignored.